state/user.go

```diff
@@ -215,6 +212,28 @@ func (st *State) RemoveUser(tag names.UserTag) error {
 		return nil
 	}
 
+	// remove the access to all the models and the current controller
+	// first query all the models for this user
+	modelsSummary, err := st.ModelSummariesForUser(tag, true)
+	if err != nil {
+		return err
+	}
+
+	for _, summary := range modelsSummary {
+		// remove all the access permissions
+		removeModelOps := removeModelUserOps(summary.UUID, tag)
+		if err := st.db().RunTransactionFor(summary.UUID, removeModelOps); err != nil {
+			logger.Errorf("impossible to remove user from models", err)
+			return err
+		}
+	}
+
+	// remove the user from the user controllers
+	if err := st.removeControllerUser(tag); err != nil {
```

Comment:
You should actually be getting this user from state first to check it still exists.